backend/src/main/kotlin/app/ehrenamtskarte/backend/stores/database/repos/AcceptingStoresRepository.kt

```diff
@@ -73,6 +76,39 @@ object AcceptingStoresRepository {
             .limit(limit, offset)
     }
 
+    fun deleteStores(acceptingStoreIds: Iterable<Int>) {
+        val contactsDelete =
+            (AcceptingStores innerJoin Contacts).slice(Contacts.id)
+                .select { AcceptingStores.id inList acceptingStoreIds }
+                .map { it[Contacts.id] }
+
+        val physicalStoresDelete =
+            (PhysicalStores innerJoin AcceptingStores).slice(PhysicalStores.id)
+                .select { AcceptingStores.id inList acceptingStoreIds }
+                .map { it[PhysicalStores.id] }
+
+        val addressesDelete =
+            ((PhysicalStores innerJoin Addresses) innerJoin AcceptingStores).slice(Addresses.id)
+                .select { AcceptingStores.id inList acceptingStoreIds }
+                .map { it[Addresses.id] }
+
+        PhysicalStores.deleteWhere {
+            id inList physicalStoresDelete
+        }
+
+        Addresses.deleteWhere {
+            id inList addressesDelete
+        }
+
+        AcceptingStores.deleteWhere {
+            id inList acceptingStoreIds
+        }
+
+        Contacts.deleteWhere {
```

Comment:
I think the default (which I think is throwing if there is a "dangling pointer" to the deleted Contact row) is more reasonable, as this would be a hint to a DB corruption.

Reply:
So it's basically better practice to not `cascade delete`? I am not really familial with best practices here.

Reply:
I don't know if there is a "best practice", but in my opinion `cascade delete` is quite an intransparent operation and it's too easy to accidentally delete data that you didn't actually want to delete.